.changeset/wicked-trains-call.md

```diff
@@ -0,0 +1,5 @@
+---
+"starlight-links-validator": patch
+---
+
+Only trim trailing index from links if  it is the full segment
```

Comment:
```suggestion
Fixes validation issues for pages ending in `index`, e.g. `module_index`.
```

Taking the liberty of editing the changeset to match [this format](https://contribute.docs.astro.build/docs-for-code-changes/changesets/#format) so we can merge this right away.